packages/appeals-service-api/src/mappers/questionnaire-submission/has-mapper.js

```diff
@@ -0,0 +1,87 @@
+class HasQuestionnaireMapper {
+	mapToPINSDataModel(journeyResponse) {
+		return [
+			{
+				questionnaire: {
+					LPACode: journeyResponse.LPACode,
+					caseReference: journeyResponse.caseReference,
+					isAppealTypeAppropriate: journeyResponse.answers['correct-appeal-type'] ? true : false,
+					doesTheDevelopmentAffectTheSettingOfAListedBuilding: journeyResponse.answers[
+						'affects-listed-building'
+					]
+						? true
+						: false,
+					affectedListedBuildings: () => {
+						if (journeyResponse.answers['affects-listed-building']) {
+							return journeyResponse.answers['listed-building-number'];
+						}
+					},
+					siteWithinGreenBelt: journeyResponse.answers['green-belt'] ? true : false,
+					howYouNotifiedPeople: journeyResponse.answers['notification-method'] ? true : false,
+					doesSiteHaveHealthAndSafetyIssues: journeyResponse.answers['safety-risks'] ? true : false,
+					healthAndSafetyIssuesDetails: () => {
+						if (journeyResponse.answers['safety-risks']) {
+							return journeyResponse.answers['affects-listed-building'];
+						}
+					},
+					doesSiteRequireInspectorAccess: journeyResponse.answers['inspector-visit-appellant']
+						? true
+						: false,
+					//journeyResponse.answers['inspector-visit-neighbour'] - should this be housed in the same property as above possibly doNeightboursAffectNeighboringSite
+					// newPlanningConditions: journeyResponse.answers['new-planning-conditions'] ? true : false, // fieldname needs clarifying
+					// journeyResponse.answers['other-ongoing-appeals'] // fieldname needs clarifying
+					nearbyCaseReferences: () => {
+						if (journeyResponse.answers['other-ongoing-appeals']) {
+							return journeyResponse.answers['other-appeal-reference'];
+						}
+					},
+					documents: [
+						{
+							notifyingParties: journeyResponse.answers['notified-who']
+						},
+						{
+							conservationAreaMap: () => {
+								if (journeyResponse.answers['conservation-area']) {
+									journeyResponse.answers['conservation-area-upload'];
+								}
+							}
+						},
+						{
+							siteNotices: () => {
+								if (journeyResponse.answers['display-site-notice']) {
+									journeyResponse.answers['upload-site-notice'];
```

Comment:
Looks good

Reply:
Just looked in the HAS journey definition and we're missing a withCondition on the associated question. Can we resolve that as part of this please?